### PR TITLE
add longtext generation benchmark

### PR DESCRIPTION
**Pull request by zhulinJulia24**
1. benchmark update
> 1. add rope_scaling_factor argument in benchmark/profile_generation.py
> 2. add longtext benchmark for profile generation
> 3. change benchmark order of kvint4 and kvint8 to make more easy for me to record result
3. oc eval config update
> 1. use opencompass hf eval configs. OC refactor hf chat config in May, which means we should update the configs. By using OC configs to avoid manual maintenance
> 2. add max-num-workers params because of OC refactor in May
4. move pipeline interface testcase out of tools test.
5. use _deps in ut workflow instead of git clone cutlass
6. add internlm/internlm-xcomposer2-4khd-7b

### CODE REVIEW

File: autotest/tools/restful/test_restful_chat_hf_turbomind.py [hunk 1]
```diff
@@ -136,7 +136,7 @@ def test_restful_chat_pr(config, common_case_config):
 @pytest.mark.parametrize('prepare_environment', [{
     'model': 'Qwen/Qwen-7B-Chat',
     'cuda_prefix': None,
-    'tp_num': 2,
+    'tp_num': 1,
```
Comment (zhulinJulia24) [inline autotest/tools/restful/test_restful_chat_hf_turbomind.py:139]: bugfix

File: .github/scripts/set_benchmark_param.sh [hunk 2]
```diff
@@ -15,9 +15,11 @@ fi
 if [[ $1 == *"llama2"* ]] || [[ $1 == *"Llama-2"* ]]
 then
     echo "MAX_ENTRY_COUNT=--cache-max-entry-count 0.95" >> "$GITHUB_ENV"
-
-else
+elif [[ $1 == *"internlm2-chat-20b"* ]]
```
Comment (zhulinJulia24) [inline .github/scripts/set_benchmark_param.sh:18]: only llama and internlm2 should set --cache-max-entry-count value. the other models use default value

**Comment by zhulinJulia24** [comment]
long text test workflow: https://github.com/zhulinJulia24/lmdeploy/actions/runs/9316759277/job/25645643448

File: .github/resources/opencompass-hf-results.json [hunk 3]
```diff
@@ -1,14 +1,14 @@
 {
     "meta-llama/Llama-2-7b-chat": {
-        "ceval": "28.3",
-        "mmlu": "35.49",
+        "ceval": "30.44",
```
Comment (lvhan028) [inline .github/resources/opencompass-hf-results.json:3]: This is the evaluation ground truth by transformers inference, isn' t it?

Comment (lvhan028) [reply .github/resources/opencompass-hf-results.json:3]: Some datasets like cmmlu, Gaokao and so on are added in `eval_opencompass_config.py`
Should the corresponding evaluation ground truth of the added datasets be updated in this json file?

File: .github/scripts/action_tools.py [hunk 4]
```diff
@@ -143,19 +143,27 @@ def evaluate(models: List[str], datasets: List[str], workspace: str):
...
+            logging.info(
+                f'Start evaluating {target_model} ...\\nn{model_cfg}\n\n')
+        else:
+            hf_model_path = target_model
+
         with open(config_path_new, 'a') as f:
             f.write(f'\ndatasets = {datasets}\n')
-            f.write(f'\nmodels = [ {target_model} ]\n')
+            if engine_type == 'hf':
+                f.write(f'\nmodels = [ *{target_model} ]\n')
```
Comment (lvhan028) [inline .github/scripts/action_tools.py:160]: what's the motivation of "*{target_model}"?

File: .github/workflows/daily_ete_test.yml [hunk 5]
```diff
@@ -48,6 +48,11 @@ on:
         description: 'Whether start a triton server api regression'
         type: boolean
         default: true
+      pipeline_regression:
+        required: true
+        description: 'Whether start a interface pipeline regression'
```
Comment (lvhan028) [inline .github/workflows/daily_ete_test.yml:53]: typo: a -> an

File: .github/workflows/daily_ete_test.yml [hunk 6]
```diff
@@ -236,13 +242,6 @@ jobs:
         run: |
           pytest autotest/tools/restful/test_restful_chat_hf_pytorch.py -m 'gpu_num_1 and not pr_test' -n 8 --alluredir=allure-results || true
           pytest autotest/tools/restful/test_restful_chat_hf_pytorch.py -m 'gpu_num_2 and not pr_test' -n 4 --alluredir=allure-results
-      - name: Test lmdeploy - interface pipeline turbomind case
```
Comment (lvhan028) [inline .github/workflows/daily_ete_test.yml:239]: What benefits can be gained by moving this snippet to the `test_pipeline` workflow?

File: autotest/tools/convert/test_convert.py [hunk 7]
```diff
@@ -42,7 +42,7 @@ def convert(config, model_case, cuda_prefix):
             dst_path, '--model-format awq --group-size 128 --trust-remote-code'
         ]),
                                      config,
-                                     model_name,
+                                     model_case,
```
Comment (lvhan028) [inline autotest/tools/convert/test_convert.py:45]: mode's mode_case?

File: .github/resources/opencompass-hf-results.json (hunk 3 above)
Comment (zhulinJulia24) [reply .github/resources/opencompass-hf-results.json:3]: > This is the evaluation ground truth by transformers inference, isn' t it?

yes

Comment (zhulinJulia24) [reply .github/resources/opencompass-hf-results.json:3]: > Some datasets like cmmlu, Gaokao and so on are added in `eval_opencompass_config.py` Should the corresponding evaluation ground truth of the added datasets be updated in this json file?

need time to run them. I will do it.

File: .github/scripts/action_tools.py (hunk 4 above)
Comment (zhulinJulia24) [reply .github/scripts/action_tools.py:160]: > what's the motivation of "*{target_model}"?
\* means use the config in read base. for hf models we use config in opencompass. So I need add * 
image

File: autotest/tools/convert/test_convert.py (hunk 7 above)
Comment (zhulinJulia24) [reply autotest/tools/convert/test_convert.py:45]: this is a bugfix, it affect tp args. Fixed in this pr.

File: .github/workflows/daily_ete_test.yml (hunk 6 above)
Comment (zhulinJulia24) [reply .github/workflows/daily_ete_test.yml:239]: > What benefits can be gained by moving this snippet to the `test_pipeline` workflow?

test tools want to cover more models's basic function.
test restful、test pipeline  want to cover more functions of this two interface by using specific models. 

Moving pipeline interface testcase out make the testcase's motivation more clear. And also can help me to manage and rerun failed functions separately. 

For example, Some model may leads test tools unstable. I can run interface testcase separately.

File: .github/workflows/daily_ete_test.yml [hunk 8]
```diff
@@ -458,10 +455,10 @@ jobs:
       - name: Install pytorch
         run: |
           python3 -m pip cache dir
-          python3 -m pip install torch==2.2.1 torchvision==0.17.1 --index-url https://download.pytorch.org/whl/cu118
+          python3 -m pip install torch==2.2.2 torchvision==0.17.2 --index-url https://download.pytorch.org/whl/cu118
```
Comment (RunningLeon) [inline .github/workflows/daily_ete_test.yml:458]: may need do update the cached `_deps`

**Review by lvhan028 — APPROVED** [review]

**Review by RunningLeon — APPROVED** [review]
LGTM

**Pull request merged**